src/ui/foundation/types.ts

```diff
@@ -0,0 +1,23 @@
+import { Command } from './SettingsProvider/types';
+
+export class CommandEventTarget extends EventTarget {
+  commands: Map<string, Command> = new Map();
+
+  public registerCommand(command: Command): void {
+    if (command.callback) {
+      this.commands.set(command.id, command);
+    }
```

Comment:
As requested, we register only commands that have a callback. `commands` is then used in the `CommandPalette` to display available commands that have a callback.

Reply:
This is better, but I don't think callbacks and command events should coexist. We can still go ahead with this an as inbetween state if you want.